packages/react-form/src/hooks/list/tests/baselist.test.tsx

```diff
@@ -797,7 +797,7 @@ describe('useBaseList', () => {
             </>
           ))}
 
-          <button type="button" onClick={onNewDefault} />
+          <button type="button" onClick={onNewDefault as any} />
```

Comment:
I chose the brute force cast because the test doesn't deal with MouseEvents and I wanted to minimize changes to the original test. I followed the example of [clickEvent() and changeEvent()](https://github.com/Shopify/quilt/blob/f96703c5c6d9338586c4de13888f21516b94f833/packages/react-form/src/hooks/list/tests/utils.tsx#L53-L56) used throughout react-forms tests. I agree it seems wonky, but I'm not sure how to avoid it without replacing `Variant[]` values with fake `React.MouseEvent` objects.